doc/source/ray-debugging.rst

```diff
@@ -0,0 +1,174 @@
+Ray Debugging
+=============
+
+Ray has a built in debugger that allows you to debug your distributed applications.
+
+.. note::
+
+    It is currently an experimental feature and under active development. Interfaces are subject to change.
+
+Getting Started
+---------------
+
+Take the following example that is modified from the Ray README:
+
+.. code-block:: python
+
+    import ray
+    ray.init()
+
+    @ray.remote
+    def f(x):
+        ray.util.pdb.set_trace()
+        return x * x
+
+    futures = [f.remote(i) for i in range(2)]
+    print(ray.get(futures))
+
+Put the program into a file named ``debugging.py`` and execute it using:
+
+.. code-block:: bash
+
+    python debugging.py
+
+
+Each of the 4 executed tasks will drop into a breakpoint when the line
+``ray.util.pdb.set_trace()`` is executed. You can attach to the debugger by running
+
+.. code-block:: bash
+
+    ray debug
+
+The ``ray debug`` command will print an output like this:
+
+.. code-block:: text
+
+    2020-11-04 15:35:50,011	INFO worker.py:672 -- Connecting to existing Ray cluster at address: 192.168.1.105:6379
+    Active breakpoints:
+    0: ray::f() | debugging.py:6
+
+    1: ray::f() | debugging.py:6
+
+    Enter breakpoint index or press enter to refresh:
+
+
+You can now enter ``0`` and hit Enter to jump to the first breakpoint. You will be dropped into PDB
+at the break point and can use the ``help`` to see the available actions. Run ``bt`` to see a backtrace
+of the execution:
```

Comment:
Can you also have a section here for "Debugger Commands"?

https://docs.python.org/3/library/pdb.html#debugger-commands

Reply:
I added a section and linked to the to that page. It probably doesn't make sense to replicate it at this point (unless we modify the comments in the future).